lib/gen_stage/broadcast_dispatcher.ex

```diff
@@ -54,17 +70,29 @@ defmodule GenStage.BroadcastDispatcher do
     {deliver_now, deliver_later, waiting} =
       split_events(events, length, waiting)
 
-    Enum.each(demands, fn {_, pid, ref} ->
-      Process.send(pid, {:"$gen_consumer", {self(), ref}, deliver_now}, [:noconnect])
+    Enum.each(demands, fn {_, pid, ref, selector} ->
+      selected = if selector, do: Enum.filter(deliver_now, selector), else: deliver_now
+      Process.send(pid, {:"$gen_consumer", {self(), ref}, selected}, [:noconnect])
     end)
 
     {:ok, deliver_later, {demands, waiting}}
   end
 
+  defp validate_selector(opts) do
+    case Keyword.get(opts, :selector) do
+      nil -> {:ok, nil}
+      selector when is_function(selector, 1) ->
+        {:ok, selector}
+      something_else ->
+        :error_logger.error_msg(':selector option must be passed a unary function and not: ~p~n', [something_else])
```

Comment:
You can `raise ArgumentError, ":selector option must be passed a unary function, got: #{inspect something_else}"`.